components/script/dom/byteteereadintorequest.rs

```diff
@@ -0,0 +1,273 @@
+    chunk: HeapBufferSource<ArrayBufferViewU8>,
+    tee_read_request: Dom<ByteTeeReadIntoRequest>,
+}
+
+impl ByteTeeReadIntoRequestMicrotask {
+    pub(crate) fn microtask_chunk_steps(&self, can_gc: CanGc) {
+        self.tee_read_request
+            .chunk_steps(self.chunk.clone(), can_gc)
+    }
+}
+
+#[dom_struct]
+pub(crate) struct ByteTeeReadIntoRequest {
+    reflector_: Reflector,
+    for_branch2: bool,
+    byob_branch: Dom<ReadableStream>,
+    other_branch: Dom<ReadableStream>,
+    #[ignore_malloc_size_of = "Rc"]
+    read_again_for_branch_1: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    read_again_for_branch_2: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    reading: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    canceled_1: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    canceled_2: Rc<Cell<bool>>,
+    #[ignore_malloc_size_of = "Rc"]
+    cancel_promise: Rc<Promise>,
+    tee_underlying_source: Dom<ByteTeeUnderlyingSource>,
+}
+impl ByteTeeReadIntoRequest {
+    #[allow(clippy::too_many_arguments)]
+    #[cfg_attr(crown, allow(crown::unrooted_must_root))]
+    pub(crate) fn new(
+        for_branch2: bool,
+        byob_branch: &ReadableStream,
+        other_branch: &ReadableStream,
+        read_again_for_branch_1: Rc<Cell<bool>>,
+        read_again_for_branch_2: Rc<Cell<bool>>,
+        reading: Rc<Cell<bool>>,
+        canceled_1: Rc<Cell<bool>>,
+        canceled_2: Rc<Cell<bool>>,
+        cancel_promise: Rc<Promise>,
+        tee_underlying_source: &ByteTeeUnderlyingSource,
+        global: &GlobalScope,
+        can_gc: CanGc,
+    ) -> DomRoot<Self> {
+        reflect_dom_object(
+            Box::new(ByteTeeReadIntoRequest {
+                reflector_: Reflector::new(),
+                for_branch2,
+                byob_branch: Dom::from_ref(byob_branch),
+                other_branch: Dom::from_ref(other_branch),
+                read_again_for_branch_1,
+                read_again_for_branch_2,
+                reading,
+                canceled_1,
+                canceled_2,
+                cancel_promise,
+                tee_underlying_source: Dom::from_ref(tee_underlying_source),
+            }),
+            global,
+            can_gc,
+        )
+    }
+
+    pub(crate) fn enqueue_chunk_steps(
+        &self,
+        global: &GlobalScope,
+        chunk: HeapBufferSource<ArrayBufferViewU8>,
+    ) {
+        // Queue a microtask to perform the following steps:
+        let byte_tee_read_request_chunk = ByteTeeReadIntoRequestMicrotask {
+            chunk,
+            tee_read_request: Dom::from_ref(self),
+        };
+        let microtask_queue = global.microtask_queue();
+        let cx = GlobalScope::get_cx();
+        microtask_queue.enqueue(
+            Microtask::ReadableStreamByteTeeReadIntoRequest(byte_tee_read_request_chunk),
+            cx,
+        );
+    }
+
+    /// <https://streams.spec.whatwg.org/#read-request-chunk-steps>
```

Comment:
https://streams.spec.whatwg.org/#ref-for-read-into-request-chunk-steps%E2%91%A0

Reply:
You need to go to the original link, and then click on the name to open this popover that gives you the references, like below. 
<img width="760" alt="Screenshot 2025-04-14 at 3 49 37 PM" src="https://github.com/user-attachments/assets/c19041e7-98ae-432d-a1c3-9b48f80cdfad" />